source/distributed/shared_tria.cc

```diff
@@ -134,6 +135,19 @@ namespace parallel
       this->update_number_cache ();
     }
 
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::
+    copy_triangulation (const dealii::Triangulation<dim, spacedim> &other_tria)
+    {
+      Assert ((dynamic_cast<const dealii::parallel::distributed::Triangulation<dim,spacedim> *>(&other_tria) == NULL),
```

Comment:
I have considered this for a long time, but I am not sure how to do it. Since ::Triangulation is the base class of all other Triangulations, `dynamic_cast` the pointer to `::Triangulation*` will never get `NULL`. Could you please give some hints?